packages/text-annotator/src/SelectionHandler.ts

```diff
@@ -212,23 +221,87 @@ export const SelectionHandler = (
      * @see https://github.com/recogito/text-annotator-js/issues/136
      */
     setTimeout(() => {
-      const sel = document.getSelection()
+      const sel = document.getSelection();
 
       // Just a click, not a selection
       if (sel?.isCollapsed && timeDifference < CLICK_TIMEOUT) {
         currentTarget = undefined;
         clickSelect();
-      } else if (currentTarget && store.getAnnotation(currentTarget.annotation)) {
-        selection.userSelect(currentTarget.annotation, evt);
+      } else if (currentTarget && currentTarget.selector.length > 0) {
+        selection.clear();
+        upsertCurrentTarget();
+        selection.userSelect(currentTarget.annotation, clonePointerEvent(evt));
       }
     });
   }
 
-  hotkeys(SELECTION_KEYS.join(','), { element: container, keydown: true, keyup: false }, (evt) => {
+  const onContextMenu = (evt: PointerEvent) => {
+    isContextMenuOpen = true;
+
+    const sel = document.getSelection();
+
+    if (sel?.isCollapsed) return;
+
+    // When selecting the initial word, Chrome Android fires `contextmenu` 
+    // before selectionChanged.
+    if (!currentTarget || currentTarget.selector.length === 0) {
+      onSelectionChange(evt);
+    }
+    
+    upsertCurrentTarget();
+
+    selection.userSelect(currentTarget.annotation, clonePointerEvent(evt));
+  }
+
+  const onKeyup = (evt: KeyboardEvent) => {
+    if (evt.key === 'Shift' && currentTarget) {
+      const sel = document.getSelection();
+
+      if (!sel.isCollapsed) {
+        selection.clear();
+        upsertCurrentTarget();
+        selection.userSelect(currentTarget.annotation, cloneKeyboardEvent(evt));
+      }
+    }
+  }
+
+  const onSelectAll = (evt: KeyboardEvent) => {
+
+    const onSelected = () => setTimeout(() => {
+      if (currentTarget?.selector.length > 0) {
+        selection.clear();
+
+        store.addAnnotation({
+          id: currentTarget.annotation,
+          bodies: [],
+          target: currentTarget
+        });
+
+        selection.userSelect(currentTarget.annotation, cloneKeyboardEvent(evt));
+      }
+      
+      document.removeEventListener('selectionchange', onSelected);
+
+      // Sigh... this needs a delay to work. But doesn't seem reliable.
+    }, 100);
+
+    // Listen to the change event that follows
+    document.addEventListener('selectionchange', onSelected);
```

Comment:
Do I understand correctly that with the delay we make the handler wait till the existing `selectionchange` listener processes the selection and create the target? So then we'll just persist it

Reply:
So here's what I could figure out: if you hit `Cmd`+`A` it will take a while until the selection is actually done. Initially, I just waited for a while in the key listener (250ms) and then stored the current selection. But that seemed really brittle. (What if the selection hasn't happened yet, and `currentTarget` is still empty.)

Therefore, I implemented the above: 
- When `Cmd`+ `A` is pressed, I'm attaching a listener to listen for the `selectionchange` event.
- Assuming that the selectionchange event would fill the `currentTarget`, I would simply create the annotation, and remove the listener again.

It turned out that this wasn't sufficient. I still had to wait a bit more. I don't think it's event order (my listener getting triggered before the one that fills the `currentTarget`). Because if it were just about the order, a wait time of 1ms should be enough? But that didn't work.